config/v1alpha1/types_image_policy.go

```diff
@@ -0,0 +1,147 @@
+	// will be used to verify the sigstore signature.
+	// the sigstore community Rekor public key(https://raw.githubusercontent.com/sigstore/root-signing/main/targets/rekor.pub)
+	// will be used to verify the sigstore signature.
+	// +optional
+	KeyData string `json:"keyData,omitempty"`
+
+	// signedIdentity specifies what image identity the signature.
+	// claims about the image.
+	// +optional
+	SignedIdentity Identity `json:"signedIdentity,omitempty"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.identityMatchPolicy) && self.identityMatchPolicy == 'ExactRepository' ? self.dockerRepository != '' : true",message="must set dockerRepository if identityMatchPolicy is ExactRepository"
+type Identity struct {
+	// identityMatchPolicy set the type of matching to be used.
+	// Valid values are "MatchRepository", "ExactRepository". When omitted, the default value is "MatchRepository".
+	// If set identityMatchPolicy to ExactRepository, then the dockerRepository must be specified.
+	// "MatchRepository" means that the identity in the signature must be in the same repository as the image identity.
+	// "ExactRepository" means that the identity in the signature must be in the same repository as a specific identity specified by "dockerRepository".
+	// +kubebuilder:validation:Enum=MatchRepository;ExactRepository
+	// +optional
+	IdentityMatchPolicy IdentityMatchPolicy `json:"identityMatchPolicy,omitempty"`
+	// dockerReference is the reference of the image identity to be matched. This field is required if identityMatchPolicy is set to "exactReference".
+	// +optional
+	DockerRepository string `json:"dockerRepository,omitempty"`
+}
+
+// IdentityMatchPolicy defines the type of matching to be used.
+type IdentityMatchPolicy string
+
+const (
+	IdentityMatchPolicyMatchRepository = "MatchRepository"
+	IdentityMatchPolicyExactRepository = "ExactRepository"
+)
+
+// +k8s:deepcopy-gen=true
+type ImagePolicyStatus struct {
+
+	// policyJSON contains the whole policy applied to the namespace
+	// which got written to disk. This includes cluster-wide policies
+	// from the `openshift-config` namespace as well.
+	PolicyJSON string `json:"policyJSON,omitempty"`
+
+	// conditions provide details on the status of the gatherer job.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []metav1.Condition `json:"conditions" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// ImagePolicyStatusConditionType is the state of the operator's reconciliation functionality.
+type ImagePolicyStatusConditionType string
+
+const (
+	// ImagePolicySuccess designates a successful application of a ContainerRuntimeConfig CR.
+	ImagePolicySuccess ImagePolicyStatusConditionType = "Success"
+
+	// ImagePolicyFailure designates a failure applying a ContainerRuntimeConfig CR.
+	ImagePolicyFailure ImagePolicyStatusConditionType = "Failure"
+)
```

Comment:
Do we need to define the actual possible condition combinations here as well or within the MCO? 

For example something like:

```go
func Success() metav1.Condition {
	return metav1.Condition{
		Type:               ImagePolicySuccess,
		Status:             metav1.ConditionTrue,
		LastTransitionTime: metav1.Now(),
		Reason:             …,
	}
}
```
We should probably also define some reasons, like:

```go
const (
	ReasonAvailable   = "Available"
	ReasonUnavailable = "Unavailable"
	ReasonCreating    = "Creating"
	ReasonDeleting    = "Deleting"
	ReasonPending     = "Pending"
	ReasonUpdating    = "Updating"
)
```

Reply:
I don't see other APIs define functions in side openshift/api, not sure if this is a proper place.
we can define something like this within MCO: https://github.com/openshift/machine-config-operator/blob/5843b7a4b27f15bc3313ce8a41ef6de792081fcb/pkg/apis/machineconfiguration.openshift.io/v1/helpers.go#L88-L95

The resource is created and deleted is handled by the API server. Inside MCO the resources will be rolled out by creating machine config. we may only need 
```
Pending: Resource exists but failed to get rendered to a machine config
Available: Everything is up to date (on every node)
Updating: Resource is being updated to a machine config
```

Reply:
Awesome, sounds good to me!